### ISSUE TITLE
Unstable GPU performance in Bomberman ULTRA + Race the sun (OpenGL)

### ISSUE THREAD

**Issue by spiraluc** [RPCS3/rpcs3#12741]
### Quick summary

When bomberman is maximalized, the RSX utilization GPU is ~70% & CPU ~60%, real HW utilization from HWmonitor is GPU 
 ~78% CPU ~63%, however emulator have only 40FPS. 
When bomberman is minimalized the real HW utilization from HWmonitor grows up to GPU ~100% & CPU ~75%, however FPS grows to 56FPS.

This is the only game that behaves like this. Other games still perform the same regardless of maximized/minimized.

Some games (for example Race the sun-NPUB31532) behave a bit similarly, the GPU is not used anywhere near to the maximum (real or RSX utilization), but the emulator cannot reach 60FPS.

You can see everything in the attached photos.

(my pc is slow so i don't know if the problem is replicable)

### Attach a log file

[RPCS3log.zip](https://github.com/RPCS3/rpcs3/files/9690740/RPCS3log.zip)

### Attach capture files for visual issues

![maximalized](https://user-images.githubusercontent.com/54077874/193417690-41cdfdbe-922b-4beb-bdaa-2323bffafea7.png)

![minimalized](https://user-images.githubusercontent.com/54077874/193417687-854af2e0-ba6b-4217-85eb-05fbe1847b6c.png)

![Race the sun](https://user-images.githubusercontent.com/54077874/193418683-3327dff7-6938-46d2-975f-35ddcf569514.png)

### System configuration

CPU: Intel core i5 2430M
GPU: NVIDIA GT525M (openGL 4.6 only) (driver v391.35)
RAM: 8GB
WIN 10 (v21H1)

**Comment by kd-11** [comment]
OpenGL window management is a black box, the driver is responsible for most of the behaviour there, not us. Unlike vulkan, there is very little you can control about how windows behave with OpenGL. The only thing we set is the size of the viewport.
What I can suggest is to create a rpcs3 profile in NVIDIA control panel and create an aggressive profile with vsync disabled and all the performance optimizations turned on. NVIDIA's OpenGL driver is already really slow compared to Vulkan/DX11/DX12 on the same hardware.

**Comment by spiraluc** [comment]
I created an aggressive profile for the GPU, but it had no noticeable effect.
 It is interesting that when I click through the menu in game, there is a lag between each click and a short drop to 0 FPS. 

Also, in the ``customization menu'' where there are only a few pictures, including a few flashing ones, the GPU utilization is low, but the FPS are extremely low.
![snapshot](https://user-images.githubusercontent.com/54077874/193425454-5c1c13f2-60a7-412a-99ef-67a77fb2bf1b.png)

**Comment by kd-11** [comment]
You seem to have the exact same laptop I had many years ago, it is buried somewhere in storage. Dell inspiron 15R n115.
The first thing with these laptops is that the NVIDIA card works as a "3D accelerator". It doesn't actually draw to the screen, the intel GPU does that. There is a mux between the 2 cards that is very finicky.
The first thing I would try is to connect to HDMI using the port on the side of the laptop, that will connect directly to the NVIDIA display and bypass intel.

**Comment by spiraluc** [comment]
You are right that intel somehow cooperates with nvidia(they even shared the same IRQ addresses). INTEL utilization in HWinfo grows a little in every game, but not much.
![Snímek](https://user-images.githubusercontent.com/54077874/193743337-56b5cacf-fdaa-416f-a401-ff4d25d6d4d7.png)

 So I tried the VGA and HDMI port, but it seems that intel cannot be bypassed. It didn't help, because these ports always connect the monitor to the intel.
![VGA](https://user-images.githubusercontent.com/54077874/193739095-1e4814c6-8125-4acf-8e55-198423975dde.png)
![HDMI](https://user-images.githubusercontent.com/54077874/193739098-c63e3a2a-72f1-453c-930f-38a242ef146b.png)

When I stop race the sun, I get some unusual errors. 
{Performance Sensor} PERF: Failed to get per core cpu usage: A counter with a negative denominator value was detected. (0x800007d6)
E 0:01:56.406768 SIG: Thread [rsx::thread] is too sleepy. Waiting for it 23740.938µs already!
·E 0:01:56.430682 SIG: Thread [rsx::thread] is too sleepy. Waiting for it 47650.705µs already!
·E 0:01:56.478172 SIG: Thread [rsx::thread] is too sleepy. Waiting for it 95132.224µs already!
[RPCS3 perf.log](https://github.com/RPCS3/rpcs3/files/9702983/RPCS3.perf.log)
Once i get error like this:
·F 0:03:23.828174 {RSX [0x00531f0]} SIG: Thread terminated due to fatal error: Unknown register name: unk
(in file C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\rpcs3\Emu\RSX\GL\GLCommonDecompiler.cpp:36[:23], in function get_varying_register_location)
[RPCS3 error.log](https://github.com/RPCS3/rpcs3/files/9702979/RPCS3.error.log)

I ran several OGL benchmark tests in program GPU Caps Viewer. NVIDIA (OGL 4.6 support) clearly won most of the test, but INTEL (OGL 3.2 support) almost always won in the OGL 3.X tests. Looks like nvidia doesn't like OGL 3.X rendering.

triangle intel
![triangle intel](https://user-images.githubusercontent.com/54077874/193741485-037ddff8-fed0-4346-8487-1216c38af2c6.png)
triangle nvidia
![triangle nvidia](https://user-images.githubusercontent.com/54077874/193741490-73272ab5-3ea7-42b8-a199-bf5cd14dbf0d.png)
RGB triangle intel
![RGB triangle intel](https://user-images.githubusercontent.com/54077874/193741521-8e9a0128-880a-47dd-a6b7-03c15859b2f7.png)
RGB triangle nvidia
![RGB triangle nvidia](https://user-images.githubusercontent.com/54077874/193741524-a41d02dc-017e-40e6-aa95-3bbeaa76e206.png)

Draw stress tests have the same FPS.

draw stress test intel
![draw stress test intel](https://user-images.githubusercontent.com/54077874/193741543-7b48364c-d82e-47e0-a190-688a9d9a49f3.png)
draw stress test nvidia
![draw stress test nvidia](https://user-images.githubusercontent.com/54077874/193741550-5df075fe-0be4-4654-934f-b276b7accfc2.png)

**Comment by spiraluc** [comment]
Intel and Nvidia cooperation probably won't be the main problem. I tried the old build (rpcs3-v0.0.12-10950-264df111_win64) and it had much higher performance everywhere in the game. Unfortunately, I don't know exactly which build caused this.

[RPCS3old.log](https://github.com/RPCS3/rpcs3/files/9739043/RPCS3old.log)
ingame old
![ingame old](https://user-images.githubusercontent.com/54077874/194704445-168f57a4-4a89-4c79-aa7b-72ba07da767d.png)
ingame new
![ingame new](https://user-images.githubusercontent.com/54077874/194704450-5ab87a86-b414-44be-8406-9b96cfaf7f6f.png)
game choose old
![game choose old](https://user-images.githubusercontent.com/54077874/194704454-dcf3686c-c75c-49e5-a634-efc42e2f8c1b.png)
game choose new
![game choose new](https://user-images.githubusercontent.com/54077874/194704460-63448132-d4d2-4c28-84ba-97b088cb27e0.png)
customization menu old
![cust  menu old](https://user-images.githubusercontent.com/54077874/194704467-d0ad98b9-9cb4-4af3-9504-525539bd4fe7.png)
customization menu new
![cust  menu new](https://user-images.githubusercontent.com/54077874/194704471-a480128c-55b8-48c2-b764-28b11d98a899.png)

**Comment by kd-11** [comment]
I'm not able to reproduce this. It is likely a CPU issue. You'll have to bisect to identify the offending commit. https://wiki.rpcs3.net/index.php?title=Help:Using_different_versions_of_RPCS3#How_to_find_the_build_that_caused_a_regression?

**Comment by spiraluc** [comment]
I tried to find the build that caused it and found out that the biggest FPS drop was caused by 'Initial Linux Aarch64 port (PPU only) #11315' (rpcs3-v0.0.20-13178-580bd2b2_win64). 

[RPCS3 old.log](https://github.com/RPCS3/rpcs3/files/9832777/RPCS3.old.log)

![game choose old](https://user-images.githubusercontent.com/54077874/197024905-a17c7b99-454c-43b0-9215-241d0da7e03c.png)

[RPCS3 new.log](https://github.com/RPCS3/rpcs3/files/9832780/RPCS3.new.log)

![game choose new](https://user-images.githubusercontent.com/54077874/197024943-0e331187-9e3e-4c67-8174-f69fd0866831.png)


(Maybe this whole problem is caused by the PPU decoder.)

**Comment by digant73** [comment]
@spiraluc is that still an issue?

**Comment by spiraluc** [comment]
I don't use this HW anymore, so solved.
